File: src/ChpokkWeb/SystemFiles/Templates/ProjectTemplates/VisualBasic/Silverlight/1049/BusinessApplication/Models/User.partial.vb
Imports System
Imports System.ComponentModel

Namespace Web

    Partial Public Class User
#Region "Делает DisplayName привязываемым"

        ''' <summary>
        ''' Переопределение метода <c>OnPropertyChanged</c>, который вызывает уведомления об изменении свойства при изменении <see cref="User.DisplayName"/>.
        ''' </summary>
        ''' <param name="e">Аргументы события изменения свойств.</param>
        Protected Overrides Sub OnPropertyChanged(ByVal e As System.ComponentModel.PropertyChangedEventArgs)
            MyBase.OnPropertyChanged(e)

            If e.PropertyName = "Name" Or e.PropertyName = "FriendlyName" Then
                Me.RaisePropertyChanged("DisplayName")
            End If
        End Sub
#End Region
    End Class
End Namespace

File: src/ChpokkWeb/SystemFiles/Templates/ProjectTemplates/CSharp/Silverlight/1028/BusinessApplication/Models/User.partial.cs
namespace $safeprojectname$.Web
{
    using System;
    using System.ComponentModel;

    /// <summary>
    /// <see cref="User"/> 類別的延伸模組。
    /// </summary>
    public partial class User
    {
        #region 將 DisplayName 設成可繫結

        /// <summary>
        /// 覆寫會於 <see cref="User.DisplayName"/> 變更時產生屬性變更通知的 <c>OnPropertyChanged</c> 方法。
        /// </summary>
        /// <param name="e">屬性變更事件引數。</param>
        protected override void OnPropertyChanged(PropertyChangedEventArgs e)
        {
            base.OnPropertyChanged(e);

            if (e.PropertyName == "Name" || e.PropertyName == "FriendlyName")
            {
                this.RaisePropertyChanged("DisplayName");
            }
        }
        #endregion
    }
}
